vllm_ascend/compilation/compiler_interface.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (c) 2025 Huawei Technologies Co., Ltd. All Rights Reserved.
+# This file is a part of the vllm-ascend project.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import Any, Callable, Optional
+
+import torch
+import torch.fx as fx
+import torch.utils._pytree as pytree
+from torch._dynamo.backends.common import aot_autograd
+from torch._inductor.utils import output_node
+from vllm.compilation.compiler_interface import CompilerInterface
+
+
+def get_dtype_from_args(args: list[Any]) -> list[torch.dtype]:
+    """
+    Extract the dtype from the kwargs dictionary.
+    """
+    dtype_list = []
+    for value in args:
+        if isinstance(value, torch.Tensor):
+            dtype_list.append(value.dtype)
+    return dtype_list
+
+
+def get_shapes_from_args(args: list[Any]) -> list[torch.Size]:
+    """
+    Extract the shapes from the kwargs dictionary.
+    """
+    shape_list = []
+    for value in args:
+        if isinstance(value, torch.Tensor):
+            shape_list.append(value.shape)
+    return shape_list
+
+
+class AscendAdaptor(CompilerInterface):
+    name = "AscendAdaptor"
+
+    def compile(
+        self,
+        graph: fx.GraphModule,
```

Comment:
Is the graph processed by AoT dispatcher before being passed here to the compiler backend?